### DISCUSSION TITLE
how to silence remote control close-tab 'No matching tabs for expression' error

### DISCUSSION THREAD

**Discussion by tg0h** [kovidgoyal/kitty#5406] [Q&A] [answered]
Is there any way to silence the error shown if no tabs match my remote control command?

<img width="1110" alt="Screenshot 2022-08-23 at 20 10 51 - kitty@2x" src="https://user-images.githubusercontent.com/7954426/186154623-66c36da8-4851-4b2b-bdc3-c9ccec4f0286.png">

I created a remote control mapping `map ctrl+cmd+w remote_control close-tab --match "not (title:wiki or state:focused)"` to close tabs that match a certain condition.

**Comment by kovidgoyal** [comment, marked as answer]
No there isnt, it should be a simple patch to add a --no-response option
to it, like there is for many other commands, PR is welcome.

**Reply by page-down** [reply]
@kovidgoyal
Currently executing kitten in kitty does not require waiting for a response and throws errors immediately, regardless of the `--no-response`.

```conf
map f1 remote_control focus-window --match "not state:focused"
```

For the OP use case, I think there should be `--ignore-no-match` and still return an error message in case of syntax errors etc.
For true remote calls, perhaps add the `kitty @ --no-response CMD-NAME` global option and deprecate the old `CMD-NAME --no-response`.
For those that don't need to return any output or error messages, but need **error codes**, perhaps add a global option `kitty @ --quiet`.
When using `map f1 remote_control --quiet CMD-NAME`, instead of printing traceback on errors, a bell etc. will be used to indicate the error.

When I try to add `--no-response` to some commands, I find that I need to duplicate some boilerplate code, which should be discussed and designed first.

**Comment by kovidgoyal** [comment]
On Wed, Aug 24, 2022 at 12:45:35AM -0700, page-down wrote:
> When I try to add `--no-response` to some commands, I find that I need to duplicate some boilerplate code, which should be discussed and designed first.

What boilerplate code? IIRC all you need to do is add --no-response to
options_spec, the code in handle_cmd will automatically not return a
response even if the actual command returns a response.

**Reply by page-down** [reply]
Yes. I was referring to the duplicate option definitions for the individual remote control commands.
I submitted the PR first and did not include other e.g. global options etc.

Do you think `map f1 remote_control` should no longer show any error messages after using the `--no-response` option?

**Comment by kovidgoyal** [comment]
On Wed, Aug 24, 2022 at 01:48:46AM -0700, page-down wrote:
> Yes. I was referring to the duplicate option definitions for the individual remote control commands.

Well not all options can have --no-response, since it makes no sense for
something like get-text or select-window or get-colors or ls. Therefore,
it cant be a global option.

> I submitted the PR first and did not include other e.g. global options etc.
> 
> Do you think `map f1 remote_control` should no longer show any error messages after using the `--no-response` option?

No I just pushed a different fix for silencing errors from
remote-control mappings.

**Reply by page-down** [reply]
There is a typo in the recent commit.

```text
map f1 !focus-window ...
```

Needs the `remote_control` action to call.
```text
map f1 remote_control !focus-window ...
```
